### PR TITLE
New Pull Request for Takeaway Challenge

### PR DESCRIPTION
**Pull request by ChukaEbi**

### CODE REVIEW

File: lib/messager.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+require 'twilio-ruby'
+require 'dotenv'
+Dotenv.load
+
+class Messager
+  # put your own credentials here
+  def initialize
+    account_sid = ENV['account_sid']
+    auth_token = ENV['auth_token']
+    @client = Twilio::REST::Client.new account_sid, auth_token
```
Comment (pitchinvasion) [inline lib/messager.rb]: def initialize(client: nil)
  @client = client || Twilio::REST::Client
end

File: lib/order.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  end
+
+  def total
+    r = @total.round(2)
+    puts "Your total is $#{r}"
+    r
+  end
+
+  def see_menu
+    @restaurant.menu
```
Comment (pitchinvasion) [inline lib/order.rb]: use delegation

Comment (pitchinvasion) [inline lib/order.rb]: wrap instance variables - WWSMD?

**Comment by tansaku** [comment]
You had your takeaway_challenge reviewed by **Moe**.
### The good points are:
* The README has been updated
* The README has instructions
* Correctly uses named subject
* Required gems (i.e. twilio-ruby) are listed in Gemfile
* All tests test real behaviour and not stubs
* The Twilio API is stubbed appropriately
* All unit tests use doubles (London style)
* All public interface elements are tested
* The solution correctly prefers classes to modules
* The Law of Demeter is respected in the code base
* The Twilio dependency can be injected appropriately
* Business and presentation logic has been cleanly separated
* The design has at least `Takeaway` and `Order` classes or equivalent
* The solution makes good use of the Ruby language, e.g., Hash.new(0) is used to simplify management of counts
* The code follows the Ruby style and coding conventions

### You should consider the following improvements:
* Personal details and tokens on GitHub
* Open Closed Principle

### Additional comments:
Awesome code, found it difficult to pick out flaws. 

Reviewee and myself strongly believed that a module should have been used for the messaging service, but apparently this was frowned upon by consensus?

**Comment by tansaku** [comment]
You had your takeaway_challenge reviewed by **Moe**.
### The good points are:
* The README has been updated
* The README has instructions
* Correctly uses named subject
* Required gems (i.e. twilio-ruby) are listed in Gemfile
* All tests test real behaviour and not stubs
* The Twilio API is stubbed appropriately
* All unit tests use doubles (London style)
* All public interface elements are tested
* The solution correctly prefers classes to modules
* The Law of Demeter is respected in the code base
* The Twilio dependency can be injected appropriately
* Business and presentation logic has been cleanly separated
* The design has at least `Takeaway` and `Order` classes or equivalent
* The solution makes good use of the Ruby language, e.g., Hash.new(0) is used to simplify management of counts
* The code follows the Ruby style and coding conventions

### You should consider the following improvements:
* Personal details and tokens on GitHub
* Open Closed Principle

### Additional comments:
Awesome code, found it difficult to pick out flaws. 

Reviewee and myself strongly believed that a module should have been used for the messaging service, but apparently this was frowned upon by consensus?


see https://github.com/makersacademy/takeaway_challenge/blob/master/docs/review.md for more details on the items above

see also this video exemplar on how a coach would approach this challenge: https://www.youtube.com/watch?v=mgiJKdH9x8c

**Pull request closed without merging**